File: BE-Task_1-Go/internal/repository/memory/seed.go
package memory

import (
	"time"

	"gotest/internal/domain/entities"
)

func Seed(store *Store, startupReference time.Time) {
	startupReference = startupReference.UTC()

	store.mu.Lock()
	defer store.mu.Unlock()

	store.agents = map[string]entities.Agent{
		"A-L1-US-001":  {ID: "A-L1-US-001", Role: entities.RoleL1Support, Region: entities.RegionUS},
		"A-L2-US-001":  {ID: "A-L2-US-001", Role: entities.RoleL2Support, Region: entities.RegionUS},
		"A-MGR-US-001": {ID: "A-MGR-US-001", Role: entities.RoleManager, Region: entities.RegionUS},
		"A-L2-EU-001":  {ID: "A-L2-EU-001", Role: entities.RoleL2Support, Region: entities.RegionEU},
	}

	store.purchases = map[string]entities.Purchase{
		"PUR-US-OTHER-001": {ID: "PUR-US-OTHER-001", Region: entities.RegionUS, Category: entities.CategoryOther, PurchasedAt: startupReference.Add(-96 * time.Hour)},
		"PUR-US-OTHER-002": {ID: "PUR-US-OTHER-002", Region: entities.RegionUS, Category: entities.CategoryOther, PurchasedAt: startupReference.Add(-12 * time.Hour)},
		"PUR-EU-OTHER-001": {ID: "PUR-EU-OTHER-001", Region: entities.RegionEU, Category: entities.CategoryOther, PurchasedAt: startupReference.Add(-36 * time.Hour)},
		"PUR-US-ELEC-001":  {ID: "PUR-US-ELEC-001", Region: entities.RegionUS, Category: entities.CategoryElectronics, PurchasedAt: startupReference.Add(-72 * time.Hour)},
		"PUR-US-ELEC-002":  {ID: "PUR-US-ELEC-002", Region: entities.RegionUS, Category: entities.CategoryElectronics, PurchasedAt: startupReference.Add(-24 * time.Hour)},
		"PUR-US-ELEC-003":  {ID: "PUR-US-ELEC-003", Region: entities.RegionUS, Category: entities.CategoryElectronics, PurchasedAt: startupReference.Add(-6 * time.Hour)},
		"PUR-EU-ELEC-001":  {ID: "PUR-EU-ELEC-001", Region: entities.RegionEU, Category: entities.CategoryElectronics, PurchasedAt: startupReference.Add(-24 * time.Hour)},
		"PUR-US-DIGI-001":  {ID: "PUR-US-DIGI-001", Region: entities.RegionUS, Category: entities.CategoryDigitalDownload, PurchasedAt: startupReference.Add(-6 * time.Hour)},
		"PUR-US-DIGI-002":  {ID: "PUR-US-DIGI-002", Region: entities.RegionUS, Category: entities.CategoryDigitalDownload, PurchasedAt: startupReference.Add(-30 * time.Hour)},
		"PUR-US-DIGI-003":  {ID: "PUR-US-DIGI-003", Region: entities.RegionUS, Category: entities.CategoryDigitalDownload, PurchasedAt: startupReference.Add(-72 * time.Hour)},
	}

	store.refunds = map[string]entities.Refund{}
}

File: BE-Task_1-Ruby/internal/repository/memory/seed.rb
module RefundBackend
  module Repository
    module Memory
      class Seed
        def initialize(store:, startup_reference:)
          @store = store
          @startup_reference = startup_reference
        end

        def call
          @store.reset!(agents: seed_agents, purchases: seed_purchases)
        end

        private

        def seed_agents
          [
            RefundBackend::Domain::Entities::Agent.new(
              agent_id: "A-L1-US-001",
              role: RefundBackend::Domain::ValueObjects::Roles::L1_SUPPORT,
              region: RefundBackend::Domain::ValueObjects::Regions::US
            ),
            RefundBackend::Domain::Entities::Agent.new(
              agent_id: "A-L2-US-001",
              role: RefundBackend::Domain::ValueObjects::Roles::L2_SUPPORT,
              region: RefundBackend::Domain::ValueObjects::Regions::US
            ),
            RefundBackend::Domain::Entities::Agent.new(
              agent_id: "A-MGR-US-001",
              role: RefundBackend::Domain::ValueObjects::Roles::MANAGER,
              region: RefundBackend::Domain::ValueObjects::Regions::US
            ),
            RefundBackend::Domain::Entities::Agent.new(
              agent_id: "A-L2-EU-001",
              role: RefundBackend::Domain::ValueObjects::Roles::L2_SUPPORT,
              region: RefundBackend::Domain::ValueObjects::Regions::EU
            )
          ]
        end

        def seed_purchases
          [
            purchase("PUR-US-OTHER-001", "US", "other", hours_ago: 96),
            purchase("PUR-US-OTHER-002", "US", "other", hours_ago: 12),
            purchase("PUR-EU-OTHER-001", "EU", "other", hours_ago: 36),
            purchase("PUR-US-ELEC-001", "US", "electronics", hours_ago: 72),
            purchase("PUR-US-ELEC-002", "US", "electronics", hours_ago: 24),
            purchase("PUR-US-ELEC-003", "US", "electronics", hours_ago: 6),
            purchase("PUR-EU-ELEC-001", "EU", "electronics", hours_ago: 24),
            purchase("PUR-US-DIGI-001", "US", "digital_download", hours_ago: 6),
            purchase("PUR-US-DIGI-002", "US", "digital_download", hours_ago: 30),
            purchase("PUR-US-DIGI-003", "US", "digital_download", hours_ago: 72)
          ]
        end

        def purchase(purchase_id, region, category, hours_ago:)
          RefundBackend::Domain::Entities::Purchase.new(
            purchase_id: purchase_id,
            region: region,
            category: category,
            purchase_timestamp: @startup_reference - (hours_ago * 3600)
          )
        end
      end
    end
  end
end
